### PR TITLE
Security updates and update deps

### PR DESCRIPTION
**Pull request by gmelillo**
List of changes included in this PR:
- Update some dependencies
- allow testing and building without docker to speed up the development
- Change pipeline to not use the intermediate images
- Introduce automatic versioning and changelog generation
- Introduce preview images coming from PRs

### CODE REVIEW

**Comment by bassco** [comment]
Manually bump the version number due to deps changes ?
I realise that we don't use version numbers, yet, and rely on commit SHAs

**Review by bassco — COMMENTED** [review]
Some queries regarding the non-rust library code changes

File: Cargo.toml [hunk 1]
```diff
@@ -3,32 +3,45 @@
 name = "dali"
 version = "1.0.10"
 authors = ["Augusto César Dias <augusto.dias@olx.com>"]
```
[the remark this answers is not in the document]
Comment (gmelillo) [reply Cargo.toml:5]: Not sure who to add to the maintainer list

File: Dockerfile [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
-ARG BASE_IMAGE=docker.pkg.github.com/olxgroup-oss/dali/base-dali
-
-FROM ${BUILD_IMAGE}:latest as build
+FROM rust:1.59.0-slim as build
 
 WORKDIR /usr/src/dali
 
+RUN apt-get update && apt install -y \
+        libvips-dev \
+        libvips \
```
[the remark this answers is not in the document]
Comment (gmelillo) [reply Dockerfile:8]: will probably pin it to `8.10.1` used by `libvips`

File: build.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// (c) Copyright 2022-2023 OLX
+
+fn main() {
+    pkg_config::Config::new().probe("vips").unwrap();
+    println!("cargo:rustc-env=RUSTFLAGS=-C target-feature=-crt-static");
```
[the remark this answers is not in the document]
Comment (gmelillo) [reply build.rs:5]: Just to configure the compiler on build time.

File: Cargo.toml [hunk 4]
```diff
@@ -3,32 +3,45 @@
...
-edition = "2018"
+edition = "2021"
```
[the remark this answers is not in the document]
Comment (gmelillo) [reply Cargo.toml:6]: Rust editions are `2015`/`2018`/`2021` :)

File: .releaserc.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          "message": "chore(release): ${nextRelease.version}\n\n${nextRelease.notes}"
+        }
+      ],
+      [
+        "@eclass/semantic-release-docker",
+        {
+          "baseImageName": "dali",
+          "registries": [
+            {
+              "url": "ghcr.io",
```
[the remark this answers is not in the document]
Comment (gmelillo) [reply .releaserc.json:43]: I think both can be used to get and push

File: Makefile [hunk 6]
```diff
@@ -1,45 +1,27 @@
...
-	$(SUDO) docker push "$(BASE_IMAGE):latest"
+docker-build:
+	docker build -t ${RUNTIME_IMAGE}:${VERSION_TAG} .
 
-build-image:
-	$(SUDO) docker build --build-arg BASE_IMAGE=$(BASE_IMAGE) --build-arg BUILD_IMAGE=$(BUILD_IMAGE) -t "$(RUNTIME_IMAGE):$(VERSION_TAG)" .
-	$(SUDO) docker tag "$(RUNTIME_IMAGE):$(VERSION_TAG)" "$(RUNTIME_IMAGE):latest"
-	$(SUDO) docker push "$(RUNTIME_IMAGE):$(VERSION_TAG)"
-	$(SUDO) docker push "$(RUNTIME_IMAGE):latest"
+docker-publish: docker-build
```
[the remark this answers is not in the document]
Comment (gmelillo) [reply Makefile:26]: line 46 of `tests.yml`

File: Cargo.toml (hunk 4 above)
Comment (bassco) [inline Cargo.toml:6]: this should be a `feat:` commit - upgrading the version?

**Review by bassco — APPROVED** [review]
Metrics scrapping sorted - nice!

**Review by bassco — APPROVED** [review]

**Comment by bassco** [comment]
Thanks for the contribution and the struggle you went through to solve these upgrades. These ideas and fixes were incorporated in to #39

**Pull request closed without merging**